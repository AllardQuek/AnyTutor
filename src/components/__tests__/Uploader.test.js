import "@testing-library/jest-dom"; // Adding here to remove linting errors
import Dropzone from "react-dropzone-uploader";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Dropzone
        onChangeStatus={jest.fn()}
        onSubmit={jest.fn()}
        accept="audio/mp3"
        maxFiles={1}
        multiple={false}
        submitButtonDisabled={(files) => files.length > 1}
        inputContent="Drop 1 Audio File of Speech"
        PreviewComponent={false} // Cannot pass props here!
        disabled={false}
        styles={{
          dropzone: {
            overflow: "auto",
            padding: "1.5rem",
            backgroundColor: "var(--dropzone-bg-color)",
          },
          dropzoneReject: {
            borderColor: "red",
            backgroundColor: "#DAA",
          },
          submitButton: {
            borderRadius: "45px",
            color: "var(--font-light-color)",
            backgroundColor: "var(--primary-color)",
            boxShadow: "0px 8px 25px rgba(0,0,0,0.1)",
            fontWeight: 500,
            letterSpacing: "1.2px",
            fontFamily: "Roboto",
          },
          preview: {},
          inputLabel: {
            color: "var(--dropzone-font-color)",
          },
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
