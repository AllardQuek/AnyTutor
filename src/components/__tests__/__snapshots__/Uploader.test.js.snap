// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dzu-dropzone"
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "backgroundColor": "var(--dropzone-bg-color)",
      "overflow": "auto",
      "padding": "1.5rem",
    }
  }
>
  <label
    className="dzu-inputLabel"
    style={
      Object {
        "color": "var(--dropzone-font-color)",
      }
    }
  >
    Drop 1 Audio File of Speech
    <input
      accept="audio/mp3"
      className="dzu-input"
      disabled={false}
      multiple={false}
      onChange={[Function]}
      type="file"
    />
  </label>
</div>
`;
